### PR TITLE
add authentication

### PR DESCRIPTION
**Pull request by thdvbr**
- added login logout form and authentication flow 
- added auth context 
- protected /create route to be seen only when authenticated

### CODE REVIEW

File: src/components/login-form/LoginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const { login } = useContext(AuthContext);
+  const initialState = {
+    email: '',
+    password: '',
+    isSubmitting: false,
+    errorMessage: null,
+  };
+
+  const [data, setData] = useState(initialState);
+
```
Comment (irina-baeva) [inline src/components/login-form/LoginForm.jsx:23]: why not react-hook-forms?

File: src/components/protected-route/ProtectedRoute.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+export default function ProtectedRoute({
+  component: Component,
```
Comment (irina-baeva) [inline src/components/protected-route/ProtectedRoute.jsx]: i would move this kind of components in folder HOC (may be inside "components/hoc")

**Review by irina-baeva — APPROVED** [review]

File: src/components/protected-route/ProtectedRoute.jsx (hunk 2 above)
Comment (thdvbr) [reply src/components/protected-route/ProtectedRoute.jsx]: moved

File: src/components/login-form/LoginForm.jsx (hunk 1 above)
Comment (thdvbr) [reply src/components/login-form/LoginForm.jsx:23]: ill do it in another branch later

**Pull request merged**